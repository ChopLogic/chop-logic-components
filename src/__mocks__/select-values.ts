import { MultiSelectValue } from 'components/inputs/multi-select/types';
import { SelectValue } from 'components/inputs/select/types';

export const SELECT_LANGUAGES: SelectValue[] = [
  {
    id: '57b5a856-bc4b-4c2f-8295-9f44434fafc7',
    label: 'English',
  },
  {
    id: 'b4b08a09-e287-400b-ba65-afd6ae1f6fd7',
    label: 'Russian',
  },
  {
    id: '641fc722-af5a-4426-8f8a-1566cc492b91',
    label: 'German',
  },
];

export const MULTI_SELECT_VALUES: MultiSelectValue[] = [
  { id: '57b5a856-bc4b-4c2f-8295-9f44434fafc7', label: 'English', selected: false },
  { id: '583e8b60-0177-4e38-a423-4b1d5d0a2236', label: 'Spanish', selected: false },
  { id: 'b20bea7b-8172-4d12-b172-d5feac02d863', label: 'French', selected: false },
  { id: '641fc722-af5a-4426-8f8a-1566cc492b91', label: 'German', selected: false },
];
