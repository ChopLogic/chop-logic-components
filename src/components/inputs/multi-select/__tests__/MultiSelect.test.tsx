import { describe, expect, it, vi } from 'vitest';
import { render, screen } from '@testing-library/react';

import ChopLogicMultiSelect from '../MultiSelect';
import { MULTI_SELECT_VALUES } from '__mocks__/select-values';

describe('ChopLogicSelect component', () => {
  const testProps = {
    id: 'language-select-id',
    name: 'language',
    label: 'Select your language',
    values: MULTI_SELECT_VALUES,
    onSelect: vi.fn(),
    placeholder: 'Not selected',
  };

  it('should render correctly', () => {
    const { asFragment } = render(<ChopLogicMultiSelect {...testProps} />);
    screen.debug();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the label', () => {
    render(<ChopLogicMultiSelect {...testProps} />);
    expect(screen.getByLabelText(testProps.label)).toBeInTheDocument();
  });

  //   it('should have accessible roles', () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     expect(screen.getByRole('combobox')).toBeInTheDocument();
  //     expect(screen.getByRole('listbox')).toBeInTheDocument();
  //   });

  //   it('could be disabled', () => {
  //     render(<ChopLogicSelect {...testProps} disabled />);
  //     expect(screen.getByRole('combobox')).toBeDisabled();
  //   });

  //   it('could be required', () => {
  //     render(<ChopLogicSelect {...testProps} required />);
  //     expect(screen.getByRole('combobox')).toHaveAttribute('aria-required', 'true');
  //   });

  //   it('should display all values as options', () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const options = screen.getAllByRole('option');
  //     expect(options).toHaveLength(LANGUAGES.length);
  //     options.forEach((option, index) => {
  //       expect(option).toHaveTextContent(LANGUAGES[index].label);
  //     });
  //   });

  //   it('should call onSelect handler', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const combobox = screen.getByRole('combobox');
  //     await userEvent.click(combobox);
  //     const option = screen.getByText(LANGUAGES[0].label);
  //     await userEvent.click(option);
  //     expect(testProps.onSelect).toHaveBeenCalledOnce();
  //   });

  //   it('should allow the user to select an option', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const combobox = screen.getByRole('combobox');
  //     expect(combobox).toHaveValue('');
  //     await userEvent.click(combobox);
  //     const option = screen.getByText(LANGUAGES[1].label);
  //     await userEvent.click(option);
  //     expect(combobox).toHaveValue(LANGUAGES[1].id);
  //   });

  //   it('should move the focus correctly on Tab press', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const combobox = screen.getByRole('combobox');
  //     await userEvent.click(combobox);

  //     const options = screen.getAllByRole('option');
  //     options[0].focus();
  //     await userEvent.tab();
  //     expect(options[1]).toHaveFocus();
  //     await userEvent.tab();
  //     expect(options[2]).toHaveFocus();
  //     await userEvent.tab();
  //     expect(options[0]).toHaveFocus();
  //   });

  //   it('should move the focus correctly on Tab press', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const combobox = screen.getByRole('combobox');
  //     await userEvent.click(combobox);

  //     const options = screen.getAllByRole('option');
  //     options[0].focus();
  //     await userEvent.tab();
  //     expect(options[1]).toHaveFocus();
  //     await userEvent.tab();
  //     expect(options[2]).toHaveFocus();
  //     await userEvent.tab();
  //     expect(options[0]).toHaveFocus();
  //   });

  //   it('pressing ArrowDown button should move focus to the next option', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const options = screen.getAllByRole('option');
  //     options[0].focus();
  //     await userEvent.keyboard('[ArrowDown]');
  //     expect(options[1]).toHaveFocus();
  //   });

  //   it('pressing ArrowUp button should move focus to the previous option', async () => {
  //     render(<ChopLogicSelect {...testProps} />);
  //     const options = screen.getAllByRole('option');
  //     options[1].focus();
  //     await userEvent.keyboard('[ArrowUp]');
  //     expect(options[0]).toHaveFocus();
  //   });
});
