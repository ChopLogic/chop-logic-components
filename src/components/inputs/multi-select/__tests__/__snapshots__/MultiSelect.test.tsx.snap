// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicMultiSelect > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-iQQCXo cJYHMl test-class"
    style="width: 400px;"
  >
    <label
      class="sc-Qotzb DuTOm"
      for="multi-select-id"
    >
      <span>
        Select your languages
      </span>
    </label>
    <button
      aria-controls="multi-select-id_dropdown"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-required="false"
      class="sc-gDpztx ggOtHE"
      id="multi-select-id"
      name="languages"
      role="combobox"
      type="button"
      value=""
    >
      <span>
        Not selected
      </span>
      <svg
        fill="currentColor"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M21.886 5.536A1.002 1.002 0 0021 5H3a1.002 1.002 0 00-.822 1.569l9 13a.998.998 0 001.644 0l9-13a.998.998 0 00.064-1.033zM12 17.243L4.908 7h14.184L12 17.243z"
        />
      </svg>
    </button>
    <ul
      class="sc-kpOvIu kwQFiY"
      id="multi-select-id_dropdown"
      role="listbox"
      tabindex="-1"
    >
      <li
        aria-selected="false"
        class="sc-icnseD sc-jMsorb drrHqd bbzhsR"
        id="57b5a856-bc4b-4c2f-8295-9f44434fafc7"
        role="option"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
            fill="currentColor"
          />
        </svg>
        <span>
          English
        </span>
      </li>
      <li
        aria-selected="false"
        class="sc-icnseD sc-jMsorb drrHqd bbzhsR"
        id="583e8b60-0177-4e38-a423-4b1d5d0a2236"
        role="option"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
            fill="currentColor"
          />
        </svg>
        <span>
          Spanish
        </span>
      </li>
      <li
        aria-selected="false"
        class="sc-icnseD sc-jMsorb drrHqd bbzhsR"
        id="b20bea7b-8172-4d12-b172-d5feac02d863"
        role="option"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
            fill="currentColor"
          />
        </svg>
        <span>
          French
        </span>
      </li>
      <li
        aria-selected="false"
        class="sc-icnseD sc-jMsorb drrHqd bbzhsR"
        id="641fc722-af5a-4426-8f8a-1566cc492b91"
        role="option"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M14 0H2C.9 0 0 .9 0 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V2c0-1.1-.9-2-2-2zm0 14H2V2h12v12z"
            fill="currentColor"
          />
        </svg>
        <span>
          German
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
